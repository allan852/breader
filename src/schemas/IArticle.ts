export default interface IArtilce {
    author: string
    // categories?: string[]
    // comments?: string
    created_at?: number
    date: string
    feed_id?: number
    time: string
    description?: string
    feed_title?: string
    // enclosures?: InterfaceFeedEnclosure[]
    guid: string
    // image?: InterfaceFeedImage
    id?: number
    index?: number
    is_starred?: number
    is_unread?: number
    is_dayfirst?: boolean
    link: string
    // origlink?: string
    // permalink?: string
    // pubdate: string
    // source?: InterfaceFeedSource
    summary: string
    title: string
    updated_at?: number
};
