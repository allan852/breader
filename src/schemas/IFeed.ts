export default interface IFeed {
    id?: number
    title?: string
    link: string
    url: string
    date?: Date
    date_time?: string
    etag?: string
    description?: string
    favicon?: string
    summary?: string
    deleted_at?: number
}
