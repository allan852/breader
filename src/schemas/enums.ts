export enum ELogicError {
  FEEDPARSER_FETCH_ERROR = 'FEEDPARSER_FETCH_ERROR',
  FEEDPARSER_NOT_FOUND = 'FEEDPARSER_NOT_FOUND',
  FEEDPARSER_WRONG_URL = 'FEEDPARSER_WRONG_URL',
  POUCHDB_EXISTS = 'POUCHDB_EXISTS',
  UNKNOWN = 'UNKNOWN',
}

export enum EMenuKey {
  ALL_ITEMS = 'ALL_ITEMS',
  STARRED_ITEMS = 'STARRED_ITEMS',
  UNREAD_ITEMS = 'UNREAD_ITEMS',
}

export enum EArticleFilter {
  STARRED = 'STARRED',
  UNREAD = 'UNREAD',
  ALL = 'ALL',
}
