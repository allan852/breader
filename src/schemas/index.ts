export * from './enums'
export * from './IArticle'
export * from './IFeed'
export * from './IReduxAction'
export * from './IReduxAsyncAction'
export * from './IReduxState'
