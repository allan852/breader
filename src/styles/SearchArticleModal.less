@import './Values.less';

.search-article-modal {
  .ant-modal-body {
    padding: 24px 24px 12px 24px;
  }
  .search-article-keywords {
    margin-bottom: 12px;
  }
  .matched-list {
    overflow: hidden;
    .list-item {
      &:first-child {
        margin-top: 16px;
      }
      &:hover {
        background-color: @list-selected-color;
      }
    }
  }
  .empty-list {
    color: @black-text-color-dark;
    height: 22px;
    padding: 0;
    text-align: center;
    line-height: 21px;
    margin-bottom: 0;
  }
}
