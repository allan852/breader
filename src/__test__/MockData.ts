import { IntlProvider } from 'react-intl'
import { messages } from '../locales'
import { IArticle, IFeed } from '../schemas'

export const intlProviderProps: IntlProvider.Props = {
  locale: 'en-US',
  messages: messages['en-US'],
}

const intlProvider = new IntlProvider(intlProviderProps, {})
export const intl = intlProvider.getChildContext().intl

export const feed: IFeed = {
  _id: 'http://www.ruanyifeng.com/blog/atom.xml',
  _rev: '5-55772bf2b4d4b74608a60ac3480334e4',
  author: 'ruanyifeng',
  categories: [],
  createTime: 1563686468437,
  deleteTime: 0,
  description: "Ruan YiFeng's Blog",
  etag: '"11d86-58e2cc1eab031"',
  favicon: 'http://www.ruanyifeng.com/favicon.ico',
  generator: 'Movable Type 5.2.2 (http://www.sixapart.com/movabletype/)',
  language: '',
  link: 'http://www.ruanyifeng.com/blog/',
  publishTime: 1563697765000,
  time: 1563697765000,
  title: '阮一峰的网络日志',
  url: 'http://www.ruanyifeng.com/blog/atom.xml',
}

export const article: IArticle = {
  _id: 'tag:www.ruanyifeng.com,2019:/blog//1.2100',
  _rev: '5-ff6fc450616071a8f7d022b7532e85aa',
  author: 'ruanyifeng',
  categories: [],
  comments: 'article comments',
  createTime: 1563697765000,
  deleteTime: 0,
  description: 'article description',
  enclosures: [],
  feedId: 'http://www.ruanyifeng.com/blog/atom.xml',
  image: '',
  isDayFirst: true,
  isStarred: false,
  isUnread: false,
  link: 'http://www.ruanyifeng.com/blog/2019/07/weekly-issue-65.html',
  originLink: '',
  publishTime: 1563697765000,
  summary: 'article summary',
  time: 1563697765000,
  title: 'article title',
}
