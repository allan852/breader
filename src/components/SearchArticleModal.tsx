import { Input, Modal } from 'antd'
import Immutable from 'immutable'
import React, { Component } from 'react'
import { Scrollbars } from 'react-custom-scrollbars'
import { injectIntl, WrappedComponentProps } from 'react-intl'
import ArticleItem from '../containers/ArticleItem'
import { IArticle } from '../schemas'
import Utils from '../utils'

import '../styles/SearchArticleModal.less'

const Search = Input.Search

export interface ISearchArticleModalOwnProps {
  visible: boolean
  onCancel: (e: any) => any
  onItemChoose: (index: number) => any
}

export interface ISearchArticleModalReduxDispatch {}

export interface ISearchArticleModalReduxState {
  articles: Immutable.List<IArticle>
}

export interface ISearchArticleModalProps
  extends ISearchArticleModalOwnProps,
    ISearchArticleModalReduxDispatch,
    ISearchArticleModalReduxState {}

export interface ISearchArticleModalState {
  readonly keywords: string
  readonly matchedArticles: Immutable.List<IArticle>
}

class SearchArticleModal extends Component<
  ISearchArticleModalProps & WrappedComponentProps,
  ISearchArticleModalState
> {
  public searchArticles: (keywords: string[]) => any
  public constructor(props: ISearchArticleModalProps & WrappedComponentProps) {
    super(props)
    this.state = {
      keywords: '',
      matchedArticles: Immutable.List<IArticle>([]),
    }
    this.searchArticles = Utils.debounce(this._searchArticles, 100)
    this.searchArticles = this.searchArticles.bind(this)
  }
  public handleSubmit = () => {
    const keywords = this.state.keywords
    const matched = this._searchArticles(keywords.split(' '))
    if (keywords) {
      this.setState({
        matchedArticles: matched,
      })
    }
  }
  public handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      keywords: e.target.value,
    })
  }
  public componentDidUpdate() {
    if (this.props.visible) {
      setImmediate(() => {
        const input: HTMLInputElement | null = document.querySelector(
          '.search-article-keywords input'
        )
        if (input) {
          input.focus()
        }
      })
    }
  }
  public render() {
    return (
      <Modal
        className="search-article-modal"
        closable={false}
        width={376}
        style={{ top: 42 }}
        visible={this.props.visible}
        footer={null}
        onCancel={this.props.onCancel}>
        <Search
          className="search-article-keywords"
          placeholder={this.props.intl.formatMessage({ id: 'keywords' })}
          value={this.state.keywords}
          onChange={this.handleChange}
          onSearch={this.handleSubmit}
        />
        <div className="matched-list">
          <Scrollbars autoHide autoHeight autoHeightMax={'calc(100vh - 164px)'}>
            {this.state.matchedArticles.map((article: IArticle) => (
              <div
                key={article._id}
                onClick={() =>
                  this.props.onItemChoose(article.index as number)
                }>
                <ArticleItem
                  author={article.author}
                  guid={article._id}
                  time={article.time}
                  feedId={article.feedId}
                  title={article.title}
                  summary={article.summary}
                  className="item-is-unread"
                />
              </div>
            ))}
          </Scrollbars>
        </div>
      </Modal>
    )
  }
  private _searchArticles(keys: string[]) {
    const len = keys.length
    if (len < 1) {
      return Immutable.List<IArticle>([])
    }
    const matched = this.props.articles.filter(
      (article: IArticle, index: number) => {
        const str = article.title + article.author + article.summary
        let i = 0
        for (; i < len; i++) {
          if (str.indexOf(keys[i]) === -1) {
            break
          }
        }
        if (i === len) {
          article.index = index
          return true
        }
        return false
      }
    )
    return matched
  }
}

export default injectIntl(SearchArticleModal)
