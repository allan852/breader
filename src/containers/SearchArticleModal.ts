import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux'
import { Dispatch } from 'redux'
import SearchArticleModal, {
  ISearchArticleModalOwnProps,
  ISearchArticleModalReduxDispatch,
  ISearchArticleModalReduxState,
} from '../components/SearchArticleModal'
import { IReduxAction, IReduxState } from '../schemas'

const mapStateToProps: MapStateToProps<
  ISearchArticleModalReduxState,
  ISearchArticleModalOwnProps,
  IReduxState
> = (state: IReduxState) => ({
  articles: state.articles.list,
})

const mapDispatchToProps: MapDispatchToProps<
  ISearchArticleModalReduxDispatch,
  ISearchArticleModalOwnProps
> = (dispatch: Dispatch<IReduxAction>) => ({})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchArticleModal)
