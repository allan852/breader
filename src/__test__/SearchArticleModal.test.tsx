import Enzyme from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import EnzymeToJson from 'enzyme-to-json'
import Immutable from 'immutable'
import React from 'react'
import SearchArticleModal, {
    ISearchArticleModalProps,
} from '../components/SearchArticleModal'
import { IArticle } from '../schemas'
import { article, intl } from './MockData'

Enzyme.configure({ adapter: new EnzymeAdapter() })

describe('SearchArticleModal Testing', () => {
    const propsMock: ISearchArticleModalProps = {
        articles: Immutable.List<IArticle>([article]),
        onCancel: jest.fn(),
        onItemChoose: jest.fn(),
        visible: true,
    }

    it('dom testing', () => {
        const wrapper = Enzyme.mount(
            <SearchArticleModal {...propsMock} />,
            {context: {intl}}
        )
        wrapper.setProps({
            visible: false,
        })
        wrapper.setProps({
            visible: true,
        })
        wrapper.update()
        wrapper.find('input').simulate('change')
        wrapper.setState({
            keywords: 'article title',
        })
        wrapper.setState({
            matchedArticles: Immutable.List<IArticle>([article]),
        })
    })

    it('snapshot testing', () => {
        const component = Enzyme.mount(
            <SearchArticleModal {...propsMock} />,
            { context: { intl } }
        )
        const tree = EnzymeToJson(component)
        expect(tree).toMatchSnapshot()
    })
})
