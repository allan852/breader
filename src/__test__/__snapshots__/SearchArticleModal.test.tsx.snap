// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchArticleModal Testing snapshot testing 1`] = `
<InjectIntl(SearchArticleModal)
  articles={
    Immutable.List [
      Object {
        "_id": "tag:www.ruanyifeng.com,2019:/blog//1.2100",
        "_rev": "5-ff6fc450616071a8f7d022b7532e85aa",
        "author": "ruanyifeng",
        "categories": Array [],
        "comments": "article comments",
        "createTime": 1563697765000,
        "deleteTime": 0,
        "description": "article description",
        "enclosures": Array [],
        "feedId": "http://www.ruanyifeng.com/blog/atom.xml",
        "image": "",
        "isDayFirst": true,
        "isStarred": false,
        "isUnread": false,
        "link": "http://www.ruanyifeng.com/blog/2019/07/weekly-issue-65.html",
        "originLink": "",
        "publishTime": 1563697765000,
        "summary": "article summary",
        "time": 1563697765000,
        "title": "article title",
      },
    ]
  }
  onCancel={[MockFunction]}
  onItemChoose={[MockFunction]}
  visible={true}
>
  <SearchArticleModal
    articles={
      Immutable.List [
        Object {
          "_id": "tag:www.ruanyifeng.com,2019:/blog//1.2100",
          "_rev": "5-ff6fc450616071a8f7d022b7532e85aa",
          "author": "ruanyifeng",
          "categories": Array [],
          "comments": "article comments",
          "createTime": 1563697765000,
          "deleteTime": 0,
          "description": "article description",
          "enclosures": Array [],
          "feedId": "http://www.ruanyifeng.com/blog/atom.xml",
          "image": "",
          "isDayFirst": true,
          "isStarred": false,
          "isUnread": false,
          "link": "http://www.ruanyifeng.com/blog/2019/07/weekly-issue-65.html",
          "originLink": "",
          "publishTime": 1563697765000,
          "summary": "article summary",
          "time": 1563697765000,
          "title": "article title",
        },
      ]
    }
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en-US",
        "messages": Object {
          "addFeed": "Add Feed",
          "allArticlesAreReadNow": "All articles are read now.",
          "delete": "Delete",
          "doYouWantSetAllArticlesBeRead": "Do you want set all articles be read?",
          "feedUrl": "Feed Url",
          "feeds": "Feeds",
          "feedsAreUpdated": "Feeds are Updated",
          "invalidFeedUrl": "Invalid Feed Url",
          "keywords": "Keywords",
          "languages": "Languages",
          "menuAllItems": "All Items",
          "menuStarred": "Starred",
          "menuSubscriptions": "Subscriptions",
          "menuUnread": "Unread",
          "noMatched": "No matched",
          "settings": "Settings",
          "somethingWrong": "Something Wrong!",
          "title": "Breader",
          "unfoundFeed": "Unfound Feed",
          "unknown": "Unknown",
        },
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    onCancel={[MockFunction]}
    onItemChoose={[MockFunction]}
    visible={true}
  >
    <Modal
      className="search-article-modal"
      closable={false}
      confirmLoading={false}
      footer={null}
      maskTransitionName="fade"
      okType="primary"
      onCancel={[MockFunction]}
      style={
        Object {
          "top": 42,
        }
      }
      transitionName="zoom"
      visible={true}
      width={376}
    >
      <Component
        className="search-article-modal"
        closable={false}
        closeIcon={
          <span
            className="ant-modal-close-x"
          >
            <Icon
              className="ant-modal-close-icon"
              type="close"
            />
          </span>
        }
        confirmLoading={false}
        footer={null}
        maskTransitionName="fade"
        okType="primary"
        onCancel={[MockFunction]}
        onClose={[Function]}
        prefixCls="ant-modal"
        style={
          Object {
            "top": 42,
          }
        }
        transitionName="zoom"
        visible={true}
        width={376}
        wrapClassName=""
      >
        <PortalWrapper
          visible={true}
        >
          <Portal
            getContainer={[Function]}
          >
            <Portal
              containerInfo={
                <div>
                  <div>
                    <div
                      class="ant-modal-mask fade-appear"
                    />
                    <div
                      class="ant-modal-wrap "
                      role="dialog"
                      tabindex="-1"
                    >
                      <div
                        class="ant-modal search-article-modal zoom-appear"
                        role="document"
                        style="top: 42px; width: 376px;"
                      >
                        <div
                          aria-hidden="true"
                          style="width: 0px; height: 0px; overflow: hidden;"
                          tabindex="0"
                        />
                        <div
                          class="ant-modal-content"
                        >
                          <div
                            class="ant-modal-body"
                          >
                            <span
                              class="ant-input-search search-article-keywords ant-input-affix-wrapper"
                            >
                              <input
                                class="ant-input"
                                placeholder="Keywords"
                                type="text"
                                value=""
                              />
                              <span
                                class="ant-input-suffix"
                              >
                                <i
                                  aria-label="icon: search"
                                  class="anticon anticon-search ant-input-search-icon"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    class=""
                                    data-icon="search"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                    />
                                  </svg>
                                </i>
                              </span>
                            </span>
                            <div
                              class="matched-list"
                            />
                          </div>
                        </div>
                        <div
                          aria-hidden="true"
                          style="width: 0px; height: 0px; overflow: hidden;"
                          tabindex="0"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              }
            >
              <Dialog
                className="search-article-modal"
                closable={false}
                closeIcon={
                  <span
                    className="ant-modal-close-x"
                  >
                    <Icon
                      className="ant-modal-close-icon"
                      type="close"
                    />
                  </span>
                }
                confirmLoading={false}
                destroyOnClose={false}
                footer={null}
                getContainer={[Function]}
                getOpenCount={[Function]}
                keyboard={true}
                mask={true}
                maskClosable={true}
                maskTransitionName="fade"
                okType="primary"
                onCancel={[MockFunction]}
                onClose={[Function]}
                prefixCls="ant-modal"
                style={
                  Object {
                    "top": 42,
                  }
                }
                transitionName="zoom"
                visible={true}
                width={376}
                wrapClassName=""
              >
                <div>
                  <Animate
                    animation={Object {}}
                    component=""
                    componentProps={Object {}}
                    key="mask"
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="visible"
                    transitionAppear={true}
                    transitionEnter={true}
                    transitionLeave={true}
                    transitionName="fade"
                  >
                    <AnimateChild
                      animation={Object {}}
                      key="mask"
                      transitionAppear={true}
                      transitionEnter={true}
                      transitionLeave={true}
                      transitionName="fade"
                    >
                      <LazyRenderBox
                        className="ant-modal-mask"
                        hiddenClassName="ant-modal-mask-hidden"
                        key="mask"
                        style={Object {}}
                        visible={true}
                      >
                        <div
                          className="ant-modal-mask"
                          style={Object {}}
                        />
                      </LazyRenderBox>
                    </AnimateChild>
                  </Animate>
                  <div
                    aria-labelledby={null}
                    className="ant-modal-wrap "
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onMouseUp={[Function]}
                    role="dialog"
                    style={
                      Object {
                        "display": null,
                      }
                    }
                    tabIndex={-1}
                  >
                    <Animate
                      animation={Object {}}
                      component=""
                      componentProps={Object {}}
                      key="dialog"
                      onAppear={[Function]}
                      onEnd={[Function]}
                      onEnter={[Function]}
                      onLeave={[Function]}
                      showProp="visible"
                      transitionAppear={true}
                      transitionEnter={true}
                      transitionLeave={true}
                      transitionName="zoom"
                    >
                      <AnimateChild
                        animation={Object {}}
                        key="dialog-element"
                        transitionAppear={true}
                        transitionEnter={true}
                        transitionLeave={true}
                        transitionName="zoom"
                      >
                        <LazyRenderBox
                          className="ant-modal search-article-modal"
                          key="dialog-element"
                          onMouseDown={[Function]}
                          role="document"
                          style={
                            Object {
                              "top": 42,
                              "width": 376,
                            }
                          }
                          visible={true}
                        >
                          <div
                            className="ant-modal search-article-modal"
                            onMouseDown={[Function]}
                            role="document"
                            style={
                              Object {
                                "top": 42,
                                "width": 376,
                              }
                            }
                          >
                            <div
                              aria-hidden="true"
                              style={
                                Object {
                                  "height": 0,
                                  "overflow": "hidden",
                                  "width": 0,
                                }
                              }
                              tabIndex={0}
                            />
                            <div
                              className="ant-modal-content"
                            >
                              <div
                                className="ant-modal-body"
                              >
                                <Search
                                  className="search-article-keywords"
                                  enterButton={false}
                                  onChange={[Function]}
                                  onSearch={[Function]}
                                  placeholder="Keywords"
                                  value=""
                                >
                                  <Input
                                    className="ant-input-search search-article-keywords"
                                    onChange={[Function]}
                                    onPressEnter={[Function]}
                                    placeholder="Keywords"
                                    prefixCls="ant-input"
                                    suffix={
                                      <Icon
                                        className="ant-input-search-icon"
                                        onClick={[Function]}
                                        type="search"
                                      />
                                    }
                                    type="text"
                                    value=""
                                  >
                                    <span
                                      className="ant-input-search search-article-keywords ant-input-affix-wrapper"
                                    >
                                      <input
                                        className="ant-input"
                                        onChange={[Function]}
                                        onKeyDown={[Function]}
                                        placeholder="Keywords"
                                        style={null}
                                        type="text"
                                        value=""
                                      />
                                      <span
                                        className="ant-input-suffix"
                                      >
                                        <Icon
                                          className="ant-input-search-icon"
                                          key="searchIcon"
                                          onClick={[Function]}
                                          type="search"
                                        >
                                          <LocaleReceiver
                                            componentName="Icon"
                                          >
                                            <i
                                              aria-label="icon: search"
                                              className="anticon anticon-search ant-input-search-icon"
                                              onClick={[Function]}
                                              tabIndex={-1}
                                            >
                                              <IconReact
                                                className=""
                                                type="search-o"
                                              >
                                                <svg
                                                  aria-hidden="true"
                                                  className=""
                                                  data-icon="search"
                                                  fill="currentColor"
                                                  focusable="false"
                                                  height="1em"
                                                  key="svg-search"
                                                  viewBox="64 64 896 896"
                                                  width="1em"
                                                >
                                                  <path
                                                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                                    key="svg-search-svg-0"
                                                  />
                                                </svg>
                                              </IconReact>
                                            </i>
                                          </LocaleReceiver>
                                        </Icon>
                                      </span>
                                    </span>
                                  </Input>
                                </Search>
                                <div
                                  className="matched-list"
                                >
                                  <Component />
                                </div>
                              </div>
                            </div>
                            <div
                              aria-hidden="true"
                              style={
                                Object {
                                  "height": 0,
                                  "overflow": "hidden",
                                  "width": 0,
                                }
                              }
                              tabIndex={0}
                            />
                          </div>
                        </LazyRenderBox>
                      </AnimateChild>
                    </Animate>
                  </div>
                </div>
              </Dialog>
            </Portal>
          </Portal>
        </PortalWrapper>
      </Component>
    </Modal>
  </SearchArticleModal>
</InjectIntl(SearchArticleModal)>
`;
